src/NUnitFramework/framework/Internal/TestExecutionContext.cs

```diff
@@ -397,6 +401,11 @@ public IPrincipal? CurrentPrincipal
         /// </summary>
         public int CurrentRepeatCount { get; set; }
 
+        /// <summary>
+        /// Hook Extension to support high level test extensions.
+        /// </summary>
+        public HookExtension? HookExtension { get; } = new();
```

Comment:
As this is likely only used by some testing code, remove the initializer.
Developers that use it must explicitly set the Hook.

Reply:
If the users get a setter method to do this, we need an additional check that they do not overwrite some already set hook extension with registered event handlers.
To avoid unnecessary object instantiations, I would propose storing the instance in a private member and instantiate on the first get access.